controller/src/main/java/io/pravega/controller/store/checkpoint/ZKCheckpointStore.java

```diff
@@ -312,7 +312,8 @@ private void updateNode(String path, byte[] data) throws CheckpointStoreExceptio
         } catch (KeeperException.NoNodeException e) {
             // Return empty list for KeeperException.NoNodeException.
             return Collections.emptyList();
-        } catch (KeeperException.ConnectionLossException | KeeperException.OperationTimeoutException e) {
+        } catch (KeeperException.ConnectionLossException | KeeperException.OperationTimeoutException
+                | KeeperException.SessionExpiredException e) {
```

Comment:
It sounds right that we should give a proper treatment to session expiration, but I have not looked close enough at this flow to know whether simply adding an OR is correct. Also, it sounds like other operations above are not handling it explicitly either. I suggest we don't do this change and create an issue for it. Ask @shiveshr to have a look.

Reply:
Without this change the tests will fail on the new BK. I don't know why this exception was changed. I'll talk with @shiveshr

Reply:
this looks right. Though should be redundant. 

Note: curator client automatically tries to reconnect internally for connection failures. However if there is a session expiry, we shutdown our curator client, which in turn results on controller instance being shutdown too. 
This is to make sure that the controller does not erroneously assume it has "lock" or is the "leader" and continues to perform other metadata updates which could happen if we were to silently create a new session if the previous session expired. 

So even though in each of the stores (like metadata store, host index store etc) we have translated sessionExpiry to connection loss exception, the underlying curator client is never asked to create a new session. 
We have a watcher on curator's connection state. And if it moves to sessionExpired, we stop controller service. 

So translating keeper.sessionExpired exception to connection loss exception makes controller's workflows treat this as a temporary connection loss as it retries/responds to the caller with connection loss failure. [Note: once session is expired, controller cannot write to zk for any metadata update. So this translation of exception is primarily targeted towards responding to caller with appropriate "connection loss" exception] 

Also, the controller instance will shutdown in the background because of session expiry.